client/components/close-on-escape/index.jsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import { noop, last, filter } from 'lodash';
+import { Component, PropTypes } from 'react';
+
+let components = [];
+
+function onKeydown( event ) {
+	if ( components.length > 0 && event.keyCode === 27 && ! isInput( event.target ) ) { // ESC
+		const component = last( components );
```

Comment:
Would it be better to pop here and get rid of the filter in `stopCloseOnEscForComponent`?